File: Time_Conversion/time_conversion.cpp
#include<bits/stdc++.h>
using namespace std;

void timeConversion(string s){
    //if time is given in AM format
    if(s.substr(8,2) == "AM"){
        //check for case when its 12AM
        if(s.substr(0,2) == "12"){

            //add 00 as the hour value and print the rest of string
            cout << "00" << s.substr(2,6);
        }

        //if hour in between 1AM to 11AM
        else{
            //print the string without any changes
            cout << s.substr(0,8);
        }
    }

    // if not in AM, then time given would be in PM
    else{
        
        //case when time is 12PM
        //no need to modify the hour value
        if(s.substr(0,2) == "12"){
            cout << "12" << s.substr(2,6);
        }

        //if time is between 1PM and 11PM
        else{
            //convert the hour count into integer
            int time = stoi(s.substr(0,2));
            //to make into 24 hour format add 12 to hour count
            time+=12;

            //display the calculated 24hour format time
            cout << time << s.substr(2,6);
        }
    }
}

int main(){
    string s;
    cin >> s;

    //calling time conversion function
    timeConversion(s);
    return 0;
}

File: Time_Conversion/time_conversion.py
def timeConversion(s):
    #splitting time
    #inp_time will store each hour, minute, seconds value as a list
    inp_time = s.split(":")

    #getting hour value
    hour = inp_time[0]

    #check if string is given in PM format
    if(s[-2:]== "PM"):
    
        #when its 12PM, no change in format
        if(hour == "12"):
            pass

        #if hour between 1PM to 11PM
        #convert in 24 hour by adding 12
        else:
            hour = str(int(hour)+12) #typecasting here

    #if time around 12AM
    #change it to 00:mm:ss
    else:
        if(hour == "12"):
            hour = "00"

    #replace first value of inp_time by hour obtained
    inp_time[0] = hour

    #conversion to string by join()
    ans = ':'.join(inp_time)

    #output until -2, it doesn't include AM and PM string
    return ans[:-2]


#main code
s = input().rstrip()
print(timeConversion(s))